src/pretix/presale/views/event.py

```diff
@@ -348,6 +348,9 @@ def get(self, request, *args, **kwargs):
             r._csp_ignore = True
             return r
 
+        if 'web' not in request.event.sales_channels:
+            raise Http404
```

Comment:
* Can you raise Http404 with a meaningful message? That should make debugging a lot easier.

* Also, doesn't this break the reseller module, which also accesses this page?

* To avoid cheating, we should probably also restrict ``CartAdd``